### ISSUE TITLE
Function `fm_unify_coords.sf` duplicated

### ISSUE THREAD

**Issue by PPaccioretti** [finnlindgren/fmesher#13]
The `fm_unify_coords.sf` function is duplicated [here](https://github.com/inlabru-org/fmesher/blob/4dbb4f15dcf81edf74db496228d00ce5763fe091/R/mesh_2d.R#L117C1-L126C2). The second definition 
raises an error when an `sf` object is used.  

``` r
library(inlabru)
#> Loading required package: fmesher
data("gorillas_sf")

fm_mesh_2d_inla(loc = gorillas_sf$nests,
                max.edge = 40)
#> Error in sf::st_sfc(x): use st_as_sfc() to extract geometries from an sf object
```

<sup>Created on 2024-03-27 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

**Comment by finnlindgren** [comment]
Thanks for finding that! The first of the duplicates was the correct one, with `sf::st_geometry()`, that `sf:::st_as_sfc.sf()` calls internally.

**Comment by finnlindgren** [comment]
Now fixed in 0.1.5.9001
